### PR TITLE
SPCTR-548 - New: Counter Block

### PR DESCRIPTION
**Pull request by imrantushar**
### Description
New Counter Block. This block is only responsible for the count number. 

### Types of changes
New Features/Block

### How has this been tested?
This Block Have 3 Layout. 
- Number Layout - Only Count Number
- Circle Layout - Count Number with Circle Progress Bar 
- Bars Layout - Count Number with Bars Progress Bar
check all start, end, and total numbers very carefully with the proper percentage working or not

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by RajkiranBagal — APPROVED** [review]

File: assets/js/uagb-counter.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		return domElement;
+	},
+
+	_inViewInit( countUp, data ){
+		const that = this
+		const callback = entries => {
+			entries.forEach( entry => {
+				const el = entry.target
+				const hasPlayed = el.hasAttribute( 'played' );  // Check if an animation has played; If played already, do mot re-trigger it.
+				// entry.isIntersecting && ! el.classList.contains( 'is-visible' )
```
Comment (bsf-zanev) [inline assets/js/uagb-counter.js:49]: @aapatre - Is this comment required in the current stage of development?

**Review by bsf-zanev — APPROVED** [review]
Updates reviewed and approved.

File: assets/js/uagb-counter.js (hunk 1 above)
Comment (aapatre) [reply assets/js/uagb-counter.js:49]: Nope, it's old comment
Will delete it

Comment (aapatre) [reply assets/js/uagb-counter.js]: Deleted the commented code

**Review by bsf-zanev — APPROVED** [review]
All changes, including the update to Number Control, have been verified.

**Review by bsf-zanev — APPROVED** [review]
Updated code reviewed.

File: src/blocks/counter/settings.js [hunk 2]
```diff
@@ -0,0 +1,1754 @@
...
+					</InspectorTab>
+					<InspectorTab { ...UAGTabs.style }>
+						{numberStylePanel}
+						{headlineStylePanel}
+						{layout === 'circle' && circleStylePanel}
+						{layout === 'bars' && barStylePanel}
+						{layout !== 'bars' && showIcon && iconImageStylePanel}
+						{spacingPanel}
+						{/* We will be releasing the box-shadow feature later due to some technical challenges with circle layout*/}
+						{/* {layout !== 'number' && boxShadowSettings} */}
```
[the remark this answers is not in the document]
Comment (aapatre) [reply src/blocks/counter/settings.js:1743]: Yes, it's required for box shadow

File: includes/blocks/counter/frontend.js.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Frontend JS File.
+ *
+ * @since 0.0.1
+ *
+ * @package UAG_PRO
```
Comment (bsf-zanev) [inline includes/blocks/counter/frontend.js.php:7]: Package and Version need to be updated.

**Review by bsf-zanev — CHANGES_REQUESTED** [review]
All New Updates have been verified, but the Package Name and Version on some PHP Files are incorrect.

Since @aapatre is unavailable today, I shall add the necessary commit.

File: includes/blocks/counter/frontend.js.php (hunk 3 above)
Comment (bsf-zanev) [reply includes/blocks/counter/frontend.js.php]: This has been updated in [commit 024ee71](https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/1365/commits/024ee713403b65aa91ca1e491e778fe272048f98)

**Review by bsf-zanev — APPROVED** [review]
All New Updates Verified and Approved.

**Review by bsf-zanev — APPROVED** [review]
All updates Verified.

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**